assets/blocks/quiz/answer-blocks/gap-fill.js

```diff
@@ -0,0 +1,65 @@
+import { RichText } from '@wordpress/block-editor';
+import { FormTokenField } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Question block gap fill answer component.
+ *
+ * @param {Object}   props
+ * @param {Object}   props.attributes
+ * @param {string}   props.attributes.textBefore   Text before the gap.
+ * @param {string}   props.attributes.textAfter    Text after the gap.
+ * @param {string[]} props.attributes.rightAnswers Right answers.
+ * @param {Function} props.setAttributes
+ * @param {boolean}  props.hasSelected             Is the question block selected.
+ */
+const GapFillAnswer = ( {
+	attributes: { textBefore, textAfter, rightAnswers },
+	setAttributes,
+	hasSelected,
+} ) => {
+	return (
+		<ul className="sensei-lms-question-block__answer sensei-lms-question-block__answer--gap-fill">
+			<li>
+				<RichText
+					placeholder={ __( 'Text before the gap', 'sensei-lms' ) }
+					value={ textBefore }
+					onChange={ ( nextValue ) =>
+						setAttributes( { before: nextValue } )
```

Comment:
This doesn't restore from attributes.

```suggestion
						setAttributes( { textBefore: nextValue } )
```